azurelinuxagent/ga/exthandlers.py

```diff
@@ -830,10 +830,10 @@ def download(self):
                 raise ExtensionDownloadError("Failed to download extension",
                                              code=ExtensionErrorCodes.PluginManifestDownloadError)
 
-        self.pkg_file = destination
+            duration = elapsed_milliseconds(begin_utc)
+            self.report_event(message="Download succeeded", duration=duration)
```

Comment:
Can you edit the `test_it_should_use_existing_extension_package_when_already_downloaded` and `test_it_should_download_and_expand_extension_package` tests to check the call_count, once you also mock the `report_event` call.

Eg: add a patch to the `report_event` function and then check the call_count. - https://github.com/Azure/WALinuxAgent/blob/3be3e1fbf2330303f76961b87d891672e847ce4e/tests/protocol/test_wire.py#L523-L543